inst/include/LocalCop/frank.hpp

```diff
@@ -31,44 +31,37 @@ namespace LocalCop {
   ///
   /// @param[in] u1 First uniform variable.
   /// @param[in] u2 Second uniform variable. 
-  /// @param[in] theta Parameter of the Frank copula with the range $R \{0}$.
-  /// @ param give_log Whether or not to return on the log scale. 
+  /// @param[in] theta Parameter of the Frank copula with the range $R \setminus \{0\}$.
+  /// @param give_log Whether or not to return on the log scale. 
   ///
   /// @return Value of the h-function.  
   template <class Type>
   Type hfrank(Type u1, Type u2, Type theta, int give_log=0) {
-    Type term1 =  exp(-theta*u1) - Type(1.0) ;
-    Type term2 =  exp(-theta*u2) - Type(1.0) ; 
-    Type term3 =  exp(-theta) - Type(1.0) ;
-    Type ans = - log(term3 / term1 + term2) ;
-    ans -= -par*(term2 + Type(1.0)) 
-    if(give_log) return ans; else return exp(ans);
+    Type term1 = exp(-theta * u1) - Type(1.0);
+    Type term2 = exp(-theta * u2) - Type(1.0);
+    Type term3 = exp(-theta) - Type(1.0);
+    Type logans = -theta * u1 + log(term2) - log(term3 + term1 * term2);
+    if(give_log) return logans; else return exp(logans);
   }
   VECTORIZE4_ttti(hfrank)
       
   /// Calculate Frank copula PDF.
   ///
   /// @param[in] u1 First uniform variable.
   /// @param[in] u2 Second uniform variable. 
-  /// @param[in] theta Parameter of the Frank copula with the range $R \{0}$.
-  /// @ param give_log Whether or not to return on the log scale. 
+  /// @param[in] theta Parameter of the Frank copula with the range $R \setminus \{0\}$.
+  /// @param give_log Whether or not to return on the log scale. 
   ///
   /// @return Value of the copula PDF. 
   template <class Type>
   Type dfrank(Type u1, Type u2, Type theta, int give_log=0) {
-    Type term1 =  exp(-theta*u1) - Type(1.0) ;
-    Type term2 =  exp(-theta*u2) - Type(1.0); 
-    Type term3 =  exp(-theta) - Type(1.0) ;
-    Type term4 = (term3 + term1 * term2) ;
-    term4 *= term4 ;
-    Type ans = log(-theta * term3) - log(term4);
-    ans -= theta* (term1 + Type(1.0)) * (term2 + Type(1.0))
-    if(give_log) return ans; else return exp(ans);
+    Type term1 = Type(1.0) - exp(-theta);
+    Type term2 = (Type(1.0) - exp(-theta * u1)) * (Type(1.0) - exp(-theta * u2));
+    Type logans = log(theta) + log(term1) - theta * (u1 + u2) - Type(2.0) * log(term1 - term2);
+    if(give_log) return logans; else return exp(logans);
   }
-  VECTORIZE4_ttti(dfrank)    
-    
-  
+  VECTORIZE4_ttti(dfrank)
 
 } // end namespace LocalCop
 
-#endif
+#endif // LOCALCOP_FRANK_HPP
```

Comment:
Please make sure to end every file with at least one empty line.  It's an old Windows vs Linux issue that doesn't matter on most modern computers.  However, at some point on an older computer you are going to get burned (i.e., file will look weird or get corrupted).  This is why GitHub marks the missing empty line for you :)